dandi/utils.py

```diff
@@ -860,3 +867,17 @@ def post_upload_size_check(path: Path, pre_check_size: int, erroring: bool) -> N
             lgr.error(msg)
         else:
             raise RuntimeError(msg)
+
+
+def joinurl(base: str, path: str) -> str:
+    """
+    Append a slash-separated ``path`` to a base URL ``base``.  The two
+    components are separated by a single slash, removing any excess slashes
+    that would be present after naïve concatenation.
+
+    If ``path`` is already an absolute URL, it is returned unchanged.
+    """
+    if path.lower().startswith(("http://", "https://")):
```

Comment:
Function name and description suggests being applicable to URL of any transport, not just https so better be smth like

```suggestion
    if re.match('[a-z]+://', path.lower()):
```

to avoid surprises.

Reply:
but also how it is different from a standard urllib's join?

```
In [3]: urllib.parse.urljoin('http://a.c/', '/1/2/3')
Out[3]: 'http://a.c/1/2/3'

In [4]: urllib.parse.urljoin('http://a.c/', 's3://1/2/3')
Out[4]: 's3://1/2/3'
```

Reply:
* The function is only meant for HTTP(S) URLs.

* It differs from `urljoin()` for URLs that have a path that doesn't end in `/`:

    ```
    >>> urljoin("https://example.com/foo/bar", "quux")
    'https://example.com/foo/quux'
    ```

    `joinurl()` would instead give `https://example.com/foo/bar/quux`.

Reply:
* then better name it `httpurljoin` or at least describe it in its description
* also please add into description  this difference in behavior as rationale

Reply:
Done: 728882f1